forc/src/utils/defaults.rs

```diff
@@ -49,26 +49,64 @@ path = "tests/harness.rs"
 }
 
 pub(crate) fn default_program() -> String {
-    r#"script;
+    r#"contract ;
 
-fn main() {
+    abi MyContract {
 
-}
-"#
-    .into()
-}
+    }
 
-pub(crate) fn default_test_program() -> String {
-    r#"
+    impl MyContract for Contract {
 
-#[tokio::test]
-async fn harness() {
-    assert_eq!(true, true);
-}
+    }
 "#
     .into()
 }
 
+pub(crate) fn default_test_program(project_name: &str) -> String {
+    format!(
+        "{}{}{}{}{}",
+        r#"
+    use fuel_tx::{ContractId, Salt};
+    use fuels_abigen_macro::abigen;
+    use fuels_contract::{contract::Contract, parameters::TxParameters};
+    use fuels_signers::util::test_helpers;
+    
+    // Load abi from json
+    abigen!(MyContract, "./"#,
+        project_name,
```

Comment:
Hmm are you sure this path is correct? Why does loading the contract require `out/debug/` in the path?

Reply:
This assumes your json abi is the root of the project, however to load the `.bin` of a compiled sway contract it outputs into `/out/`, and by default `/out/debug/` like rust does, and then and release compile would be in a different subdir of out

Reply:
I think the user should be able to do:
`forc init <proj_name` -> `forc build` -> `forc test` without having to change any files, and everything should just work/pass without any issues. I feel like whichever paths get us there should be used.

Reply:
+1 on the running those 3 commands should not fail. Note that by default both the `-abi.json` and `.bin` are placed in `out/debug/`.

Reply:
`-abi.json` isn't generated anywhere by default is it? I thought it just came from the stdout of `forc json-abi`? Or are you suggesting `forc init` by default outputs it into `out/debug/`?

Reply:
`forc build` now produces the JSON ABI by default! #796

Reply:
Oh cool, that is definitely very helpful. I am testing locally and it doesn't appear to generate a `-abi.json` file if the abi in main.sw is empty, so `forc build` doesn't make the needed file for `forc test` to work (I have checked forc version). Would you maybe want a shell function in main.sw then or would it be expected someone would add an abi to the contract before trying to test?

Reply:
This is something that's hard to know until you actually use it and suffer with it 😂... but my gut feeling is to have the template include a simple ABI with one method (after all, what use is a contract without an ABI?).

Reply:
I added a simple test_function method to the abi, and to the contract (as ABI must be implemented by the Contract) that returns true, and now everything works in 3 commands!